File: api/app/Models/District.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class District extends Model
{
    protected $fillable = ['province_id', 'name', 'code', 'is_active'];

    protected $casts = [
        'is_active' => 'boolean',
    ];

    public function province(): BelongsTo
    {
        return $this->belongsTo(Province::class, 'province_id');
    }

    public function customers(): HasMany
    {
        return $this->hasMany(Customer::class, 'district_id');
    }

    public function orders(): HasMany
    {
        return $this->hasMany(Order::class, 'district_id');
    }
}

File: api-dotnet/AdminHiitop.Api/Domain/Catalog/Entities/District.cs
using AdminHiitop.Api.Domain.Common;
using AdminHiitop.Api.Domain.Sales.Entities;

namespace AdminHiitop.Api.Domain.Catalog.Entities;

public sealed class District : AuditableEntity
{
    public int ProvinceId { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;

    public Province Province { get; set; } = null!;
    public ICollection<Customer> Customers { get; set; } = new List<Customer>();
    public ICollection<Order> Orders { get; set; } = new List<Order>();
    public ICollection<Warehouse> Warehouses { get; set; } = new List<Warehouse>();
}
